core/notifications/src/main/java/com/google/samples/apps/nowinandroid/core/notifications/AndroidSystemNotifier.kt

```diff
@@ -16,17 +16,152 @@
+                    .setGroup(NEWS_NOTIFICATION_GROUP)
+                    .setAutoCancel(true)
+            }
+        }
+        val summaryNotification = newsNotification {
+            val title = getString(
+                R.string.news_notification_group_summary,
+                newsNotifications.size,
+            )
+            setContentTitle(title)
+                .setContentText(title)
+                .setSmallIcon(
+                    com.google.samples.apps.nowinandroid.core.common.R.drawable.ic_nia_notification,
+                )
+                // Build summary info into InboxStyle template.
+                .setStyle(newsInboxStyle(newsResources, title))
+                .setGroup(NEWS_NOTIFICATION_GROUP)
+                .setGroupSummary(true)
+                .setAutoCancel(true)
+                .build()
+        }
+
+        with(NotificationManagerCompat.from(this)) {
+            newsNotifications.forEachIndexed { index, notification ->
+                notify(newsResources[index].id.hashCode(), notification)
+            }
+            notify(NEWS_NOTIFICATION_SUMMARY_ID, summaryNotification)
+        }
     }
+
+    /**
+     * Creates an inbox style summary notification for news updates
+     */
+    private fun newsInboxStyle(
+        newsResources: List<NewsResource>,
+        title: String,
+    ): InboxStyle = newsResources
+        // Show at most 5 lines
+        .take(5)
+        .fold(InboxStyle()) { inboxStyle, newsResource ->
+            inboxStyle.addLine(newsResource.title)
+        }
+        .setBigContentTitle(title)
+        .setSummaryText(title)
+}
+
+/**
+ * Creates a notification for configured for news updates
+ */
+private fun Context.newsNotification(
+    block: NotificationCompat.Builder.() -> Unit,
+): Notification {
+    ensureNotificationChannel()
+    return NotificationCompat.Builder(
+        this,
+        NEWS_NOTIFICATION_CHANNEL_ID,
+    )
+        .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+        .apply(block)
+        .build()
 }
+
+/**
+ * Ensures the a notification channel is is present if applicable
+ */
+private fun Context.ensureNotificationChannel() {
+    if (VERSION.SDK_INT < VERSION_CODES.O) return
+
+    val channel = NotificationChannel(
+        NEWS_NOTIFICATION_CHANNEL_ID,
+        getString(R.string.news_notification_channel_name),
+        NotificationManager.IMPORTANCE_DEFAULT,
+    ).apply {
+        description = getString(R.string.news_notification_channel_description)
+    }
+    // Register the channel with the system
+    NotificationManagerCompat.from(this).createNotificationChannel(channel)
+}
+
+private fun Context.newsPendingIntent(newsResource: NewsResource): PendingIntent? =
+    PendingIntent.getActivity(
+        this,
+        NEWS_NOTIFICATION_REQUEST_CODE,
+        // TODO: Read color from material theme to style the chrome custom tab
+        //  this is currently only readable from composition.
+        CustomTabsIntent.Builder()
```

Comment:
I did this, but the experience of opening the app first, then linking externally is jarring. Would you be okay with keeping the current behavior but also having a pending intent that marks the task as read?

Reply:
Going back to launching the app, and starting the chrome custom tab because of [notification trampoline restrictions](https://developer.android.com/about/versions/12/behavior-changes-12#notification-trampolines). We should consider moving into an internal app webview because of this, launching 2 activities so quickly is really ugly.